app/src/main/java/nl/mpcjanssen/simpletask/Simpletask.kt

```diff
@@ -454,17 +441,17 @@ class Simpletask : ThemedNoActionBarActivity() {
                 val selectedTasks = TodoList.selectedTasks
                 val initialCompleteTasks = ArrayList<Task>()
                 val initialIncompleteTasks = ArrayList<Task>()
-                var cbState: Boolean?
-                cbState = selectedTasks.getOrNull(0)?.isCompleted()
 
-                selectedTasks.forEach {
-                    if (it.isCompleted()) {
-                        initialCompleteTasks.add(it)
-                        if (!(cbState ?: false)) { cbState = null }
+                val first : Boolean? = selectedTasks.getOrNull(0)?.isCompleted()
+
+                val cbState: Boolean? = selectedTasks.fold(first) { stateSoFar, task ->
+                    val completed = task.isCompleted()
+                    if (completed) {
+                        initialCompleteTasks.add(task)
                     } else {
-                        initialIncompleteTasks.add(it)
-                        if (cbState ?: true) { cbState = null }
+                        initialIncompleteTasks.add(task)
                     }
+                    stateSoFar?.takeIf { it == completed }
```

Comment:
Most of this is style. A little is for clarity, but I could accomplish a similar thing with a smaller change.